modules/src/clients/ics13_near/client_def.rs

```diff
@@ -56,31 +58,79 @@ impl<T: HostFunctionsProvider> ClientDef for NearClient<T> {
         &self,
         _ctx: &dyn ReaderContext,
         _client_id: ClientId,
-        client_state: Self::ClientState,
+        mut client_state: Self::ClientState,
         header: Self::Header,
     ) -> Result<(), Error> {
-        // your light client, shouldn't do storage anymore, it should just do verification here.
-        validate_light_block::<T>(&header, client_state)
+        // TODO: optimize the verification, according to https://nomicon.io/ChainSpec/LightClient
+        for header in header.inner {
```

Comment:
this isn't using batch proofs via the block merkle root. Thought this was already implemented?

Reply:
I decided to postpone it for now, just to make sure everything works without them. Should I work on it in this PR?

Reply:
no need, lets just create a tracking issue. So we can get to integration testing quickly.

Reply:
Sounds good